FlowCrypt/src/test/java/com/flowcrypt/email/security/pgp/PgpMsgTest.kt

```diff
@@ -0,0 +1,217 @@
+      ),
+
+      MessageInfo(
+        key = "decrypt - [enigmail] encrypted iso-2022-jp pgp-mime",
+        content = listOf("=E3=82=BE=E3=81=97=E9=80=B8=E7=8F=BE"), // part of "ゾし逸現飲"
+        charset = "ISO-2022-JP",
+      ),
+
+      MessageInfo(
+        key = "decrypt - [enigmail] encrypted iso-2022-jp, plain text",
+        content = listOf(
+          // complete string "ゾし逸現飲"
+          decodeString("=E3=82=BE=E3=81=97=E9=80=B8=E7=8F=BE=E9=A3=B2", "UTF-8")
+        ),
+        charset = "ISO-2022-JP",
+      )
+    )
+
+    private fun decodeString(s: String, charset: String): String {
+      val bytes = s.substring(1).split('=').map { Integer.parseInt(it, 16).toByte() }.toByteArray()
+      return String(bytes, Charset.forName(charset))
+    }
+
+    private fun loadResource(path: String): ByteArray {
+      return PgpMsgTest::class.java.classLoader!!
+        .getResourceAsStream("${PgpMsgTest::class.simpleName}/$path")
+        .readBytes()
+    }
+
+    private fun loadResourceAsString(path: String): String {
+      return String(loadResource(path), StandardCharsets.UTF_8)
+    }
+
+    private fun loadSecretKey(file: String): PGPSecretKeyRing {
+      return PGPainless.readKeyRing().secretKeyRing(loadResourceAsString("keys/$file"))
+    }
+
+    private fun loadMessage(file: String): String {
+      return loadResourceAsString("messages/$file")
+    }
+
+    private fun findMessage(key: String): MessageInfo {
+      return messages.firstOrNull { it.key == key }
+        ?: throw IllegalArgumentException("Message '$key' not found")
+    }
+  }
+
+  @Test // ok
+  fun multipleDecryptionTest() {
+    val keys = listOf(
+      "decrypt - without a subject",
+      "decrypt - [enigmail] encrypted iso-2022-jp pgp-mime",
+      "decrypt - [enigmail] encrypted iso-2022-jp, plain text",
+      "decrypt - [gpg] signed fully armored message"
+    )
+    for (key in keys) {
+      println("Decrypt: '$key'")
+      val r = processMessage(key)
+      assertTrue("Message not returned", r.content != null)
+      val messageInfo = findMessage(key)
+      checkContent(
+        expected = messageInfo.content,
+        actual = r.content!!.toByteArray(),
+        charset = messageInfo.charset
+      )
+    }
+  }
+
+  @Test // ok
+  fun decryptionTest1() {
+    val r = processMessage("decrypt - [security] mdc - missing - error")
+    assertTrue("Message is returned when should not", r.content == null)
+    assertTrue("Error not returned", r.error != null)
+    assertTrue("Missing MDC not detected", r.error!!.type == PgpMsg.DecryptionErrorType.NO_MDC)
+  }
+
+  @Test // ok
+  fun decryptionTest2() {
+    val r = processMessage("decrypt - [security] mdc - modification detected - error")
+    assertTrue("Message is returned when should not", r.content == null)
+    assertTrue("Error not returned", r.error != null)
+    assertTrue("Bad MDC not detected", r.error!!.type == PgpMsg.DecryptionErrorType.BAD_MDC)
+  }
+
```

Comment:
Is there a test somewhere to test:
 - wrong key provided
 - wrong pass phrase provided
 - key missing pass phrase

Also, please name the existing test more appropriately alike `missingMdcProducesError` and `modifiedMdcProducesError`